inc/Engine/Preload/Controller/CheckExcludedTrait.php

```diff
@@ -34,34 +37,179 @@ public function add_cache_reject_uri_to_excluded( array $regexes ): array {
 		$url     = user_trailingslashit( $url );
 
 		foreach ( $regexes as $regex ) {
 			if ( ! is_string( $regex ) ) {
 				continue;
 			}
 
-			$regex = user_trailingslashit( $regex );
+			if ( $sanitize ) {
+				$regex = $this->format_url( $regex );
+				$regex = str_replace( '?', '\\?', $regex );
+			}
 
-			if ( preg_match( "@$regex$@m", $url ) ) {
+			if ( preg_match( "@$regex@m", $url ) ) {
 				return true;
 			}
 		}
 		return false;
 	}
+
+	/**
+	 * Indicate if we need to exclude the url.
+	 *
+	 * @param string $url url to check.
+	 * @return bool
+	 */
+	protected function is_url_excluded( string $url ): bool {
+		$queries = wp_parse_url( $url, PHP_URL_QUERY );
+
+		if ( ! $queries ) {
+			return false;
+		}
+
+		$query_array = [];
+
+		parse_str( $queries, $query_array );
+
+		$excluded_queries = rocket_get_ignored_parameters();
+
+		return count( array_intersect( array_keys( $query_array ), array_keys( $excluded_queries ) ) ) > 0 || $this->is_excluded_by_filter( $url );
+	}
+
+	/**
+	 * Check if the URL has query string.
+	 *
+	 * @param string $url URL to check.
+	 *
+	 * @return bool
+	 */
+	public function has_query_string( string $url ) {
+		$queries = wp_parse_url( $url, PHP_URL_QUERY ) ?: '';
+
+		if ( ! $queries ) {
+			return false;
+		}
+
+		$query_array = [];
+
+		parse_str( $queries, $query_array );
+
+		$query_array = $this->drop_excluded_params( array_keys( $query_array ), true );
+
+		return ! empty( $query_array );
+	}
+
+	/**
+	 * Check if the url has query params.
+	 *
+	 * @param string $url url to check.
+	 * @return bool
+	 */
+	public function has_cached_query_string( string $url ) {
+		$queries = wp_parse_url( $url, PHP_URL_QUERY ) ?: '';
+
+		if ( empty( $queries ) ) {
+			return true;
+		}
+
+		$queries = $this->convert_query_to_array( $queries );
+
+		$queries = $this->drop_excluded_params( $queries );
+
+		$cache_query_string = get_rocket_cache_query_string();
+
+		$default_params = [
+			'lang',
+			'permalink_name',
+			'lp-variation-id',
+		];
+
+		/**
+		 * At this point we’re in the WP’s search page.
+		 * This filter allows to cache search results.
+		 *
+		 * @since 2.3.8
+		 *
+		 * @param bool $cache_search True will force caching search results.
+		 */
+		if ( apply_filters( 'rocket_cache_search', false ) ) {
+			$default_params [] = 's';
+		}
+
+		$cache_query_string = array_merge(
+			$cache_query_string,
+			$default_params
+		);
+
+		if ( ! $cache_query_string ) {
+			return count( $queries ) === 0;
+		}
+
+		return empty( $queries ) || count( array_intersect( array_keys( $queries ), $cache_query_string ) ) > 0;
+	}
+
+	/**
+	 * Drop excluded query params.
+	 *
+	 * @param array $queries queries from the url.
+	 * @param bool  $only_keys has the query array only keys.
+	 *
+	 * @return array|int[]|string[]
+	 */
+	protected function drop_excluded_params( array $queries, bool $only_keys = false ) {
+		$ignored_queries = apply_filters( 'rocket_cache_ignored_parameters', [] );
```

Comment:
```suggestion
		$ignored_queries = rocket_get_ignored_parameters();
```

use rocket_get_ignored_parameters function instead of the filter directly in all instances.